### ISSUE TITLE
Pico CMS app delivers .svg types with text Content-Type

### ISSUE THREAD

**Issue by bernd-wechner** [nextcloud/cms_pico#143]
Can't include .svg images in Pico CMS .md files because the app returns them with:
```
Content-Disposition: attachment; filename="myfile.svg"
Content-type: text/plain;charset=UTF-8
```
response headers. At least that's what it looks like. They convert into HTML using a relative link tot he asset folder, but if I fetch it with `curl -i` I can see why they won't render and it's the two headers above.

It's typically up the webserver to set these headers and I may find a work around there:

https://css-tricks.com/snippets/htaccess/serve-svg-correct-content-type/

But this response is generated by `index.php` with the wrong Content-Type.

**Comment by bernd-wechner** [comment]
So yeah, I could work around in my webserver with a post request response header reset based on the URL match .svg$. But that isn't the ideal solution. It will likely work into perpetuity ;-) but in reality it's possible to respond to URLs ending .svg without delivering an SVG file! This is still a response header that the Nextcloud app should be setting properly for SVG assets.

**Comment by PhrozenByte** [comment]
See https://github.com/nextcloud/cms_pico/pull/99#issuecomment-617168973

**Comment by bernd-wechner** [comment]
Yeah seems conservatism around stuff like this forces some webserver solutions. I already have the server fixing the CSP header and now the CT header for .svg$ URLs  in response. What this recurrently encountered conservatism fails to take into account is that there are NextClouds and there are Nextclouds and it would be nice if a default level of conservatism were configurable by local admins. Why? Because I have a Nextcloud for one person, I have a Nextcloud for a family, and I have one for a club (8 users) and there are Nextclouds for many 10s of or 100s users even and all have very very different security concerns in the user space. I have zero problem for example in the space of a PcioCMS website managed for the club that maybe 2 or 3 office holders access to allow SVG files served, to allow a good few other things that are hobbled with intent in PicoCMS for Nextcloud always on this pretext of security which takes a conservative approach to the user space (does not accommodate for NextCloud instances with small trusted user bases. But that's life, we have response tweaking options at the server side, and other options ;-). Just be nice if Netxcloud apps in general considered more configurable security than they tend to. The Forms app refuses to allow HTML in the Form description intro (though I tweaked it to allow it locally) on this same pretext. Because it's a security concern. And it sure is in a context where the user base is large, anonymous, and or simply not trusted.  But that is not the context of all Netxclouds and their admins. All a digression. This can probably close under the same conservatism. Perhaps worth my filing a General Nextcloud issue (enhancement suggestion) for a setting that is akin to "Trusted user base" with a long description of "Allow users to include raw HTML in context it is normally prohibited for security reasons - use with caution, only if you have a small trusted user base" and then Nextcloud and app writers can refer tot his when clamping down on SVG, raw HTML and other content clamp downs for security. Just a thought and digression.

**Comment by PhrozenByte** [comment]
Security is a big concern because often features can indeed be used in a safely manner (like SVGs), but it can't be guaranteed, because it depends on other stuff. A lot of users work like the following:

> Hmm, feature XY isn't working as I expected... Oh, there's a checkbox? There's some tech gibberish about security and stuff next to it... No idea what it means... Let's try anyway. Oh wow, it works now! Let's leave it that way.

I don't claim that this is true for yourself! But unfortunately a lot of people out there work exactly like that. Thus we must ensure security. This is unfortunate, but there's simply no other solution. Sorry for the inconvenience.

For your particular use case you could use (vanilla) Pico CMS + Nextcloud. Remove the Pico CMS for Nextcloud app, install vanilla [Pico](http://picocms.org/) on your webserver and add Pico's `content` directory as local storage to your Nextcloud. There's no website management and no additional features besides editing Markdown files, but also no security features limiting your options. However, please keep in mind that this is **unsupported** and **a huge security risk**, especially if you allow other users to edit contents. **Don't do this if you really know what you do!**

**Comment by bernd-wechner** [comment]
Did some background reading in the interim, and it seems SVG does have a speckled history and nest security concerns being an open XML format that supports an internal DOM and scripting.

That said most references I found table two categories of security risk, one being script/CSS injection and the other DOS attacks  (it's easy to write an SVG that brings the client to it knees). The former seem to center on trust exploitation. Injected code is as secure as any JS (which is effectively sandboxed in the browser if I understand correctly) so JS/CSS injection risks center around control of what's presented to users, and ability to listen in and retransmit any of their inputs, so standard phishing tools and sniffing opportunities. All meaning indeed, that loading an SV into the client DOM would demand the SVG be vetted by a trusted supplier if you will.

That said, if used in a IMG tag, JS injection doesn't work (I tried), not sure about CSS. But of course the server doesn't off hand know the context of the request and if I load an SVG in the address bar then JS injection does work. So it looks like SVG is at the the subject of security concerns and evolutions as both sides of the issue converse (web designed wanting vecotor graphics and web service providers who need security and to provide trustworthy and trusted services).

All good leaning tanks. But what I do see some utility in, albeit it demand some infrastructure, is the ability to audit and sign SVG files. I guess that's coming some day if the community wants to use vector graphics in browsers. In the mean time, converting to PNG at different scales is the best option it seems ....
